OfflineLoginRealmAccess/OfflineLoginRealmAccess/ContentView.swift

```diff
@@ -0,0 +1,208 @@
+                    }
+                    .disabled(username.isEmpty || password.isEmpty)
+                }
+                Spacer()
+                VStack {
+                    Text(appAuthManager.signUpMessage)
+                        .foregroundStyle(.green)
+                    Text(appAuthManager.errorMessage)
+                        .foregroundStyle(.red)
+                }
+                Spacer()
+            }
+
+            if appAuthManager.isLoading {
+                ProgressView()
+            }
+        }
+    }
+}
+
+struct BookLibraryView: View {
+    var user: User
+    @ObservedResults(Book.self) var books
+
+    @State private var text: String = ""
+    @State private var showingAlert = false
+    @State private var isLoading: Bool = false
+
+    var body: some View {
+        NavigationStack {
+            ZStack {
+                List(books) { book in
+                    NavigationLink(value: book) {
+                        HStack {
+                            Image(systemName: "person.crop.circle")
+                              .foregroundColor(.white)
+                              .background((book.isPublic ? Color.green : Color.red))
+                              .clipShape(Circle())
+                            VStack(alignment: .leading) {
+                                Text(book.title)
+                                Text(book.author)
+                                    .foregroundStyle(.blue)
+                            }
+                            Spacer()
+                            VStack(alignment: .trailing) {
+                                Text(book.releaseDate, format: .dateTime.year())
+                            }
+                        }
+                        .frame(height: 50)
+                    }
+                }
+                .navigationDestination(for: Book.self) { book in
+                    BookEditView(book: book)
+                }
+            }
+            .navigationTitle("My List")
+            .toolbar {
+                ToolbarItemGroup(placement: .navigationBarLeading) {
+                    Button("Log Out") {
+
+                        realmApp.currentUser?.logOut { _ in }
+                    }
+                }
+                ToolbarItemGroup(placement: .navigationBarTrailing) {
+                    Button("Add Book") {
+                        showingAlert.toggle()
+                    }
+                    .alert("Book title", isPresented: $showingAlert) {
+                        TextField("Book's title", text: $text)
+                        Button("Create", action: addNewBook)
+                    } message: {
+                        Text("Title of the book.")
+                    }
+                }
+            }
+        }
+    }
+
+    func addNewBook() {
+        let newBook = Book()
+        newBook.title = text
+        newBook.userId = user.id
+        newBook.isPublic = false
+        $books.append(newBook)
+    }
```

Comment:
If I go to add more than one book, the second time the alert pops up, it still has the title from the first book. We probably need to clear the `$text` value in `addNewBook`.

```suggestion
        $books.append(newBook)
        $text.wrappedValue = ""
    }
```